app/uk/gov/hmrc/gform/obligation/ObligationService.scala

```diff
@@ -18,56 +18,87 @@ package uk.gov.hmrc.gform.obligation
 import java.text.SimpleDateFormat
 
 import cats.data.NonEmptyList
-import cats.kernel.Eq
-import cats.implicits._
 import uk.gov.hmrc.gform.auth.AuthService
 import uk.gov.hmrc.gform.auth.models.MaterialisedRetrievals
 import uk.gov.hmrc.gform.controllers.helpers.FormDataHelpers
 import uk.gov.hmrc.gform.gformbackend.GformConnector
-import uk.gov.hmrc.gform.sharedmodel._
-import uk.gov.hmrc.gform.sharedmodel.form.{Form, FormId, UserData}
+import uk.gov.hmrc.gform.sharedmodel.{ NotChecked, Obligations, _ }
+import uk.gov.hmrc.gform.sharedmodel.form.{ Form, FormId, UserData }
 import uk.gov.hmrc.gform.sharedmodel.formtemplate._
 import uk.gov.hmrc.http.HeaderCarrier
+import cats.implicits._
 
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.{ ExecutionContext, Future }
 
 class ObligationService(gformConnector: GformConnector) {
 
   val stringToDate = new SimpleDateFormat("yyyy-MM-dd")
 
-  private def lookupObligationsMultiple(
+  private def makeAllInfoList(
+    id: HmrcTaxPeriod,
+    evaluatedId: HmrcTaxPeriodWithEvaluatedId,
+    obligation: List[ObligationDetail]) =
+    obligation.map(
+      i =>
+        TaxPeriodInformation(
+          id,
+          evaluatedId.idNumberValue,
+          i.inboundCorrespondenceFromDate,
+          i.inboundCorrespondenceToDate,
+          i.periodKey))
+
+  private def updatedObligations(
+    idNumbers: NonEmptyList[HmrcTaxPeriodWithEvaluatedId],
+    taxResponses: List[TaxResponse]): List[TaxPeriodInformation] =
+    taxResponses.flatMap(j =>
```

Comment:
avoid the use of letters to represent objects, good names are can represent something related about the object youre dealing with at hand.. instead of "j" you could yous taxResponse or tResp something more meaningful.

"idNumbers.find(x"   x does not mean much and its HmrcTaxPeriodWithEvaluatedId. find something short that can identify this well.